// dist/controllers/model/book.js
'use strict';

// Declare datas of books
var BookList = require('./connection');

var _require = require('./connection'),
    find = _require.find;
/* const BookList = [
  { id: 1, name: 'Book 1', status: 'Available' },
  { id: 2, name: 'Book 2', status: 'Available' },
  { id: 3, name: 'Book 3', status: 'Unavailable' },
  { id: 4, name: 'Book 4', status: 'Available' },
  { id: 5, name: 'Book 5', status: 'Unavailable' },
  { id: 6, name: 'Book 6', status: 'Available' },
  { id: 7, name: 'Book 7', status: 'Unavailable' },
];
 */

console.log(find.result);

// Export datas from books get, and add new book
module.exports.find = function (callback) {
  callback(undefined, BookList); // Get all books
};

module.exports.save = function (book, callback) {
  var _book = book,
      name = _book.name,
      status = _book.status;

  if (!name || !status) {
    callback({ message: 'Book is invalid' });
  }

  book = {
    id: Date.now(), // Get id by time now
    name: name,
    status: status
  };
  BookList.push(book);
  callback(undefined, book);
};

// Export data books by Id: get, delete and update
// Get by id
module.exports.findById = function (id, callback) {
  callback(undefined, BookList.find(function (item) {
    return item.id == id;
  })); // Get book by Id
};

// Delete book by id
module.exports.delete = function (id, callback) {
  var roweffected = BookList.length;
  BookList.splice(id - 1, 1);
  roweffected -= BookList.length;
  callback(undefined, { roweffected: roweffected });
};

// Update book by id
module.exports.update = function (id, book, callback) {
  var oldBook = BookList.find(function (item) {
    return item.id == id;
  });
  if (!oldBook) {
    callback('Book not found');
    return;
  }
  var index = BookList.indexOf(oldBook);
  Object.assign(oldBook, book); // Set data oldBook by book
  BookList.fill(oldBook, index, ++index); // Change oldBook at index position by oldBook (new)
  callback(undefined, oldBook);
};
